internal/v4/api_test.go

```diff
@@ -681,36 +683,45 @@ func (s *APISuite) TestMetaPerm(c *gc.C) {
 		},
 	})
 
-	// Publish one of the revisions to development and check that the development ACLs
-	// have changed.
-	err = s.store.Publish(newResolvedURL("~charmers/precise/wordpress-23", 23), mongodoc.DevelopmentChannel)
+	// Publish one of the revisions to development, then PUT to meta/perm
+	// and check that the development ACLs have changed.
+	err := s.store.Publish(newResolvedURL("~charmers/precise/wordpress-23", 23), mongodoc.DevelopmentChannel)
 	c.Assert(err, gc.IsNil)
-	s.assertPut(c, "precise/wordpress-23/meta/perm/read", []string{"bob", "charlie"})
 
-	httptesting.AssertJSONCall(c, httptesting.JSONCallParams{
-		Handler: s.srv,
-		Do:      bakeryDo(nil),
-		URL:     storeURL("precise/wordpress-23/meta/perm"),
-		ExpectBody: params.PermResponse{
-			Read:  []string{"bob", "charlie"},
-			Write: []string{"charmers"},
-		},
+	s.doAsUser("bob", func() {
+		// Check that we aren't allowed to put to the newly published entity as bob.
+		s.assertPutUnauthorized(c, "~charmers/precise/wordpress/meta/perm/read?channel=development", []string{}, `unauthorized: access denied for user "bob"`)
```

Comment:
As mentioned below, I think the function name is ambiguous. perhaps assertPutNotAuthorized would be better, or perhaps assertPutStatusUnauthorized or possibly make the function assertPutError and make http.StatusUnauthorized a parameter.